xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/DecimalField.java

```diff
@@ -46,4 +50,46 @@ public void setValue(Object value) throws InvalidOutcomeException {
 
         super.setData(value.toString());
     }
+
+    @Override
+    public void setNgDynamicFormsValidators(JSONObject validators) {
+        //locale specific separators could be used, but it should be based on the locale of the browser
+        //char separator = new DecimalFormatSymbols(Locale.getDefault(Locale.Category.FORMAT)).getDecimalSeparator();
+        char separator = Gateway.getProperties().getString("Webui.decimal.separator", ".").charAt(0);
+
+        super.setNgDynamicFormsValidators(validators);
+
+        if (StringUtils.isNotBlank(pattern)) {
+            validators.put("pattern", pattern);
+        }
+        else {
+            Integer totalDigits = null;
+            Integer fractionDigits = null;
+
+            if(contentType.hasFacet(Facet.TOTALDIGITS)) {
+                Facet f = contentType.getFacet(Facet.TOTALDIGITS);
+                totalDigits = Integer.valueOf(f.getValue());
+            }
+
+            if(contentType.hasFacet(Facet.FRACTIONDIGITS)) {
+                Facet f = contentType.getFacet(Facet.FRACTIONDIGITS);
+                fractionDigits = Integer.valueOf(f.getValue());
+            }
+
+            if (totalDigits == null && fractionDigits == null) {
+                //default validator for any decimal field
+                validators.put("pattern", "^\\d+\\" + separator + "?\\d+$");
+            }
+            else if (totalDigits != null) {
+                if (fractionDigits == null) {
+                    validators.put("pattern", "^\\d{0," + totalDigits + "}$");
+                }
+                else
+                    validators.put("pattern", "^\\d{0," + (totalDigits - fractionDigits) + "}\\" + separator + "?\\d{0," + fractionDigits + "}$");
+            }
+            else {
+                validators.put("pattern", "^\\d+\\" + separator + "?\\d{0," + fractionDigits + "}$");

```

Comment:
Use the pattern I suggested to enforce the exact number of decimals as specified in the user requirements.